linkerd/proxy/http/src/overwrite_authority.rs

```diff
@@ -0,0 +1,157 @@
+use super::h1;
+use futures::{try_ready, Future, Poll};
+use http;
+use http::header::HOST;
+use http::header::{HeaderValue, IntoHeaderName};
+use http::uri::Authority;
+use tracing::trace;
+
+pub trait ExtractAuthority<T> {
+    fn extract(&self, target: &T) -> Option<Authority>;
+}
+
+pub trait ForceAbsForm {
+    fn is_abs_form(&self) -> bool;
+}
```

Comment:
If we called this `ShouldOverwriteHeader`, would that change anything semantically? I'm still not 100% comfortable with this being so closely relate to absolute-form.